enhancements/windows-containers/disconnected-support.md

```diff
@@ -0,0 +1,355 @@
+cluster install time or at some point during the cluster's lifetime. When mirror settings are updated, WMCO will react by:
+- creating a new set of hosts.toml files
+- copying over the new registry config files to each istance, if a file has changed
+
+All changes detailed in this enhancement proposal will be limited to the Windows Machine Config Operator and its 
+sub-component, Windows Instance Config Daemon (WICD).
+
+Example: A cluster contains the following resources
+```yaml
+apiVersion: config.openshift.io/v1 
+kind: ImageDigestMirrorSet 
+metadata:
+  name: ubi9repo
+spec:
+  imageDigestMirrors: 
+  - mirrors:
+    - example.io/example/ubi-minimal
+    - example.com/example2/ubi-minimal
+    source: registry.access.redhat.com/ubi9/ubi-minimal 
+    mirrorSourcePolicy: AllowContactingSource 
+  - mirrors:
+    - mirror.example.com
+    source: registry.redhat.io
+    mirrorSourcePolicy: NeverContactSource
+```
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: ImageTagMirrorSet
+metadata:
+  name: tag-mirror
+spec:
+  imageTagMirrors:
+  - mirrors:
+    - docker.io
+    source: docker-mirror.internal
+    mirrorSourcePolicy: AllowContactingSource
+```
+
+would result in WMCO creating 3 files to place on each Windows instance.
+```powershell
+$ tree $config_path
+C:/k/containerd/registries/
+|── registry.access.redhat.com
+|   └── hosts.toml
+|── mirror.example.com
+|   └── hosts.toml
+└── docker.io
+    └── hosts.toml
+
+
+$ cat "$config_path"/registry.access.redhat.com/host.toml
+server = "https://registry.access.redhat.com" # default fallback server since "AllowContactingSource" mirrorSourcePolicy is set
+
+[host."https://example.io/example/ubi-minimal"]
+ capabilities = ["pull"]
+
+[host."https://example.com/example2/ubi-minimal"] # secondary mirror
+ capabilities = ["pull"]
+
+
+$ cat "$config_path"/registry.redhat.io/host.toml
+# "server" omitted since "NeverContactSource" mirrorSourcePolicy is set
+
+[host."https://mirror.example.com"]
+ capabilities = ["pull"]
+
+
+$ cat "$config_path"/docker.io/host.toml
+server = "https://docker.io"
+
+[host."https://docker-mirror.internal"]
+ capabilities = ["pull", "resolve"] # resolve tags
+```
+
+As on the Linux side, it is upon the user to ensure there is no clash between mirrors across their IDMS and ITMS resources.
+
+---
+
+### Test Plan & Infrastructure Needed
+
+In addition to unit testing individual WMCO/WICD packages and controllers, an e2e job will be added to the release repo
+for WMCO's master/release-4.15 branches. A new CI workflow will be created leveraging 
+[existing step-registry steps](https://github.com/openshift/release/tree/master/ci-operator/step-registry/aws/provision/vpc/disconnected),
+which creates a disconnected cluster in AWS or vSphere with hybrid-overlay networking. 
+We'll also need to pre-load the images used in our e2e suite into the cluster's internal registry as part of job setup.
```

Comment:
Is this possible?

Reply:
I have not tried but we should be able to use `oc mirror` or `skopeo` commands to copy over images into the internal registry at install time

Reply:
Are you sure the internal registry supports hosting Window images? This needs to be clarified in the design section.

Reply:
In the past the only issue was the foreign layer links, but now those are removed and images carry the full load when pushed to a registry. I asked the image registry folks to make sure, waiting on a response

Reply:
Confirmed with the image registry folks (and a customer) that OCI compliant Windows images can be hosted fine by the internal registry, existing OCP docs should suffice on how to do this